lib/p5.fab.js

```diff
@@ -3,239 +3,250 @@ let _once = false;
-    let g = this.immediateMode.geometry;
-
-    this._savedShapesCount = this._savedShapesCount + 1 || 0;
-    g.gid = "saved|" + this._savedShapesCount; // assign gid to cache buffer
-    g._makeTriangleEdges = function () {
-        return this;
-    }; // shadow this function to avoid loosing edges when `model(...)` is called
-
-    // assign a new geometry to immediateMode to avoid pointer aliasing
-    this.immediateMode.geometry = new p5.Geometry();
-
-    return g;
+  if (this.immediateMode.shapeMode !== 0x0000)
+    // POINTS
+    this._processVertices(...arguments);
+  this.isBezier = false;
+  this.isQuadratic = false;
+  this.isCurve = false;
+  this.immediateMode._bezierVertex.length = 0;
+  this.immediateMode._quadraticVertex.length = 0;
+  this.immediateMode._curveVertex.length = 0;
+
+  // patch and return geometry
+  let g = this.immediateMode.geometry;
+
+  this._savedShapesCount = this._savedShapesCount + 1 || 0;
+  g.gid = "saved|" + this._savedShapesCount; // assign gid to cache buffer
+  g._makeTriangleEdges = function () {
+    return this;
+  }; // shadow this function to avoid loosing edges when `model(...)` is called
+
+  // assign a new geometry to immediateMode to avoid pointer aliasing
+  this.immediateMode.geometry = new p5.Geometry();
+
+  return g;
 };
 
 p5.prototype.saveShape = function () {
-    if (this._renderer.isP3D) {
-        return this._renderer.saveShape(...arguments);
-    } else {
-        console.warn("Don't use saveShape in 2D mode.");
-    }
+  if (this._renderer.isP3D) {
+    return this._renderer.saveShape(...arguments);
+  } else {
+    console.warn("Don't use saveShape in 2D mode.");
+  }
 };
 
 /*
 call fabDraw once, immediately after setup and before first draw()
 pre is called before every draw, so we wrap predraw in a closure to run only once
 */
 p5.prototype.predraw = (function (b) {
-    console.log('registered');
-
-    return function () {
-        if (!_once) {
-            _once = true;
-            if (typeof fabDraw === "function") {
-                _fab.model = ''; // clear last model
-                _fab.commands = []; // clear the gcode commands -- commandStream is preserved
-                fabDraw();
-                _fab.parseGcode();
-                _syncVizStream = true; // new model needs to be synced after current print job
-
-            }
-        }
-    };
+  console.log("registered");
+
+  return function () {
+    if (!_once) {
+      _once = true;
+      if (typeof fabDraw === "function") {
+        _fab.model = ""; // clear last model
+        _fab.commands = []; // clear the gcode commands -- commandStream is preserved
+        fabDraw();
+        _fab.parseGcode();
+        _syncVizStream = true; // new model needs to be synced after current print job
+      }
+    }
+  };
 })();
 
-p5.prototype.registerMethod('pre', p5.prototype.predraw);
-
+p5.prototype.registerMethod("pre", p5.prototype.predraw);
+
+const defaultPrinterSettings = {
```

Comment:
extracted from the constructor. I changed `filament/nozzleRadius` to `filament/nozzleDiameter` to match the json configuration and general specs of printers